### PR TITLE
Make all host references case insensitive.

### PR DESCRIPTION
**Pull request by pcarinhas**
Fixes ZPS-3989

Set all hostrefs to case insensitive testing for any hostname.

### CODE REVIEW

File: ZenPacks/zenoss/OpenStackInfrastructure/hostmap.py [hunk 1]
```diff
@@ -113,9 +121,12 @@ def add_hostref(self, hostref, source="Unknown"):
         else:
             log.debug("Tracking hostref %s (source=%s)", hostref, source)
 
-        self.mapping[hostref] = None
+        self.mapping[hostref.lower()] = None
```
Comment (joshw) [inline ZenPacks/zenoss/OpenStackInfrastructure/hostmap.py]: don't need this here- it's already been normalized above

File: ZenPacks/zenoss/OpenStackInfrastructure/hostmap.py [hunk 2]
```diff
@@ -113,9 +121,12 @@ def add_hostref(self, hostref, source="Unknown"):
...
 
     def assert_host_id(self, hostref, hostid):
+        hostref = normalize_hostname(hostref)
+        if not hostref:
```
Comment (joshw) [inline ZenPacks/zenoss/OpenStackInfrastructure/hostmap.py]: let's skip this

File: ZenPacks/zenoss/OpenStackInfrastructure/hostmap.py [hunk 3]
```diff
@@ -230,7 +244,7 @@ def perform_mapping(self):
                     hostref, clean_hostref,
                     source="Cleaned name vs original name", oneway=True)
 
-            naive_mapping[hostref] = prepId("host-{0}".format(clean_hostref))
+            naive_mapping[hostref.lower()] = prepId("host-{0}".format(clean_hostref))
```
Comment (joshw) [inline ZenPacks/zenoss/OpenStackInfrastructure/hostmap.py]: I think by this point they will already be lowercase as well.

File: ZenPacks/zenoss/OpenStackInfrastructure/hostmap.py [hunk 4]
```diff
@@ -240,7 +254,7 @@ def perform_mapping(self):
 
             naive_hostid = naive_mapping[hostref]
             log.debug("Potential host mapping: %s -> %s", hostref, naive_hostid)
-            new_mapping[hostref] = naive_hostid
+            new_mapping[hostref.lower()] = naive_hostid
```
Comment (joshw) [inline ZenPacks/zenoss/OpenStackInfrastructure/hostmap.py]: ditto

**Comment by pcarinhas** [comment]
Made all changes as requested.. Squashed.

**Pull request merged**